packages/wrangler/src/autoconfig/details.ts

```diff
@@ -118,23 +124,73 @@ export async function getDetailsForAutoConfig({
 		packageJson,
 		buildCommand: detectedFramework?.buildCommand ?? packageJsonBuild,
 		outputDir: detectedFramework?.dist ?? (await findAssetsDir(projectPath)),
+		workerName: toValidWorkerName(packageJson?.name ?? basename(projectPath)),
 	};
 }
 
+const invalidWorkerNameCharsRegex = /[^a-z0-9-]/g;
+const invalidWorkerNameStartEndRegex = /^-|-$/g;
+
+function checkWorkerNameValidity(
```

Comment:
I don't know that we should be overly prescriptive here... what if these requirements change? What do C3 and `wrangler init` do?

Reply:
Cool, sounds good. If possible, can you find docs links confirming these limits and add them as comments?

Reply:
done 🙂 

https://github.com/cloudflare/workers-sdk/pull/11187/commits/1bd21df17cef0c2e783604aac98a96333bd84e3a